tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,682 @@
+  style Agent fill:#e1f5fe,stroke:#0288d1
+
+---
+
+### Workflow Diagram
+
+#### Update-Changelog (mgmt-plane)
+
+```mermaid
+flowchart TD
+      A[Start: UpdateChangelogAsync] --> B[Validate packagePath] 
+      B --> C{packagePath valid?}
+      C -->|No| D[Return failure: Package path required/invalid]
+      C -->|Yes| E[Discover SDK repository root] 
+    
+      E --> F{Repository root found?}
+      F -->|No| G[Return failure: Failed to discover SDK repo]
+      F -->|Yes| H[Get update-changelog-script configuration] 
+    
+      H --> I{Configuration retrieval successful?}
+      I -->|No| J[Return failure: Failed to get update-changelog-script configuration] 
+      I -->|Yes| K{Configuration type?}
+    
+      K -->|Command| L[Prepare update-changelog command] 
+      K -->|ScriptPath| M[Prepare update-changelog script] 
+    
+      L --> N[Execute update-changelog command/script] 
+      M --> N
+    
+      N --> O{update-changelog process completed?} 
+      O -->|Failed| P[Return failure with exit code and output]
+      O -->|Success| Q[Return success]
+    
+      style A fill:#e1f5fe
+      style Q fill:#c8e6c9
+      style D fill:#ffcdd2
+      style G fill:#ffcdd2
+      style J fill:#ffcdd2
+      style P fill:#ffcdd2
+```
+
+> **Note:** Other tools (version update, metadata update, CI update) follow a similar flowchart pattern.
+
+---
+
+## Open Questions
+
+---
+
+## Success Criteria
+
+This tool set is complete when:
+
+- [ ] All four tools emit structured JSON with consistent schema.
+- [ ] Management-plane package run updates version + changelog automatically.
+- [ ] Data-plane run returns advisory for changelog and version.
+- [ ] Cross-language scripts execute without manual intervention.
+- [ ] Errors surface actionable mitigation text.
+
+---
+
+## Agent Prompts
+
+### Scenario 1: Update Management-Plane Package (existing SDK) for Release
+
+**Prompt:**
+
+```text
+Update the package at /home/dev/sdk/healthdataaiservices/Azure.ResourceManager.HealthDataAIServices for a stable release.
+```
+
+**Expected Agent Activity:**
+
+1. Run changelog update (auto-generation script).
+2. Run version update with inferred `--release-type stable`.
+3. Run metadata update (README, API export, formatting if needed) and summarize JSON results.
+
+### Scenario 2: Update Version for Stable Release (Data-Plane)
+
+**Prompt:**
+
+```text
+Update the version for package at /repo/sdk/healthdataaiservices/Azure.Health.Deidentification.
+```
+
+**Expected Agent Activity:**
+
+1. Confirm with user on the version to use.
+2. Run version update with the confirmed version.
+3. Report the result.
+
+### Scenario 3: Create CI Config for New SDK
+
+**Prompt:**
+
+```text
+Add initial CI configuration for a new SDK at /work/sdk/new-service/new-package.
+```
+
+**Expected Agent Activity:**
+
+1. Run CI update tool.
+2. Report the created configuration file.
+3. Suggest running update-changelog tool next.
+
+---
+
+## CLI Commands
```

Comment:
nit: next-steps is a property in the response not a part of the message https://github.com/Azure/azure-sdk-tools/blob/06abfe4918dafe5ba5c6b4f7251b33589992ae65/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Models/Responses/CommandResponse.cs#L43